### ISSUE TITLE
./run_test.py --test chainer-doc seems not to run all doctest

### ISSUE THREAD

**Issue by sonots** [chainer/chainer-test#343]
From: https://github.com/chainer/chainer/pull/3653#issuecomment-338873868

I noticed that `./run_test.py --test chainer-doc --gpu-id 1` of chainer-test repo does not test all:

```
Doctest summary
===============
  151 tests
    0 failures in tests
    0 failures in setup code
    0 failures in cleanup code
build succeeded
```

so I checked manually as:

```
$ python setup.py install develop --user && cd docs && make doctest
...
Doctest summary
===============
  896 tests
    0 failures in tests
    0 failures in setup code
    0 failures in cleanup code
```

**Comment by sonots** [comment]
it seems that `run_test.py` does not run tests of chainer functions, etc:

```
Document: reference/generated/chainer.functions.stack
-----------------------------------------------------
1 items passed all tests:
  17 tests in default
17 tests in 1 items.
17 passed and 0 failed.
Test passed.

Document: reference/generated/chainer.functions.broadcast
---------------------------------------------------------
1 items passed all tests:
   6 tests in default
6 tests in 1 items.
6 passed and 0 failed.
Test passed.
```

are not appeared in run_test.py

**Comment by sonots** [comment]
It seems cupy-doc is fine.

**Comment by sonots** [comment]
```
cd .. && python setup.py install develop --user && cd docs && make html && make clean && make doctest

Doctest summary
===============
  151 tests
    0 failures in tests
    0 failures in setup code
    0 failures in cleanup code
build succeeded
```

(this is the way ./run_test.py does)

```
cd .. && python setup.py install develop --user && cd docs && make html && make doctest

File "tutorial/type_check.rst", line 117, in default
Failed example:
    utils.type_check.expect(x_type.ndim == 2)
Expected:
    Traceback (most recent call last):
    ...
    chainer.utils.type_check.InvalidType: Expect: in_types[0].ndim == 2
    Actual: 3 != 2
Got:
    Traceback (most recent call last):
      File "/home/sonots/.pyenv/versions/3.5.3/lib/python3.5/doctest.py", line 1321, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        utils.type_check.expect(x_type.ndim == 2)
      File "/home/sonots/chainer-test/chainer/chainer/utils/type_check.py", line 514, in expect
        expr.expect()
      File "/home/sonots/chainer-test/chainer/chainer/utils/type_check.py", line 472, in expect
        '{0} {1} {2}'.format(left, self.inv, right))
    chainer.utils.type_check.InvalidType:
    Invalid operation is performed in: Reshape (Forward)

    Expect: in_types[0].ndim == 2
    Actual: 3 != 2
```

or

```
Doctest summary
===============
  896 tests
    0 failures in tests
    0 failures in setup code
    0 failures in cleanup code
```

(it looks "Expected" and "Actual" are mixed, and treated mistakenly. it sometimes succeeds)

**Comment by sonots** [comment]
doctest on theano_function.py fails.

**Comment by kmaehashi** [comment]
This issue was fixed via #356.
I get the same results from local run and Jenkins run.
